### ISSUE TITLE
Cannot cast JaxrsHttpFacade to OIDCHttpFacade in keycloak-dropwizard v1.1.0

### ISSUE THREAD

**Issue by chris-downs** [ahus1/keycloak-dropwizard-integration#25]
When verifying bearer tokens I'm running into a ClassCastException when Keycloak's BearerTokenRequestAuthenticator tries to cast keycloak-dropwizard's JaxrsHttpFacade to OIDCHttpFacade. It looks like this is due to the implementation of the base HttpFacade instead of OIDCHttpFacade. Here is the stacktrace: 

```
2019-03-11 14:43:28 DEBUG org.keycloak.adapters.PreAuthActionsHandler:75 - adminRequest http://localhost:8080/xxxx/xxxx/xxxx
2019-03-11 14:43:28 DEBUG org.keycloak.adapters.RequestAuthenticator:118 - NOT_ATTEMPTED: bearer only
ERROR [2019-03-11 21:43:28,297] io.dropwizard.jersey.errors.LoggingExceptionMapper: Error handling a request: e84fbd5fed91762f
! java.lang.ClassCastException: Cannot cast de.ahus1.keycloak.dropwizard.JaxrsHttpFacade to org.keycloak.adapters.OIDCHttpFacade
! at java.base/java.lang.Class.cast(Class.java:3606)
! at org.keycloak.adapters.BearerTokenRequestAuthenticator$2.challenge(BearerTokenRequestAuthenticator.java:182)
! at de.ahus1.keycloak.dropwizard.KeycloakAuthFilter.validateRequest(KeycloakAuthFilter.java:118)
! at de.ahus1.keycloak.dropwizard.KeycloakAuthFilter.filter(KeycloakAuthFilter.java:54)
! at org.glassfish.jersey.server.ContainerFilteringStage.apply(ContainerFilteringStage.java:132)
...
```

If you look at keycloak's [JaxrsHttpFacade implementation](https://github.com/keycloak/keycloak/blob/4.8.3.Final/adapters/oidc/jaxrs-oauth-client/src/main/java/org/keycloak/jaxrs/JaxrsHttpFacade.java#L41) in keycloak-jaxrs-oauth-client it implements OIDCHttpFacade. I think this is just a matter of implementing that class instead of the base class (or maybe even just using the JaxrsHttpFacade provided in keycloak's jaxrs client if you aren't doing anything special in yours).

keycloak-dropwizard version: 1.1.0
Keycloak version: 4.8.3.Final
Java version: OpenJDK 11.0.1

**Comment by chris-downs** [comment]
I guess my suggestion assumes that everyone is using OIDC, which might not be the case. This makes just changing the base class the wrong solution.

**Comment by ahus1** [comment]
It seems that you are the first to use the policy enforcer with this plugin. I've pushed a test case and a code change. Do you want to give it a try if this works for you (could you build yourself a snapshot version?)

If this works for you, I'll publish a proper release.

I know this will only work with OIDC, but all unit tests use only OIDC. if this doesn't work for someone, I'll find a solution once someone complains.

**Comment by chris-downs** [comment]
Finally got around to building a snapshot and can confirm that everything works in master. Thanks for the quick turnaround! 

PS: I dont think a comment fix warrants a whole new issue, but i think you mean `@Auth` instead of `@User` in your example here: https://github.com/ahus1/keycloak-dropwizard-integration/blob/b5c86156814b0f9eee3bc10291ca263439514f67/keycloak-dropwizard-jaxrs-example/src/main/java/de/ahus1/lottery/adapter/dropwizard/LotteryApplication.java#L42

**Comment by ahus1** [comment]
I've published version 1.1.1 that includes this fix. It should be available on maven central soon. 

For future changes to try out there's now also a snapshot repository available with prebuilt versions to try out. If you encounter problems, please submit them in this ticket or in a new ticket. 

A test case that shows the problem or even a pull request with a change is always welcome.
